master/internal/resourcemanagers/resource_pool.go

```diff
@@ -60,16 +65,22 @@ func NewResourcePool(
 		scheduler:     scheduler,
 		fittingMethod: fittingMethod,
 
-		agents:      make(map[*actor.Ref]*agentState),
-		taskList:    newTaskList(),
-		groups:      make(map[*actor.Ref]*group),
-		scalingInfo: &sproto.ScalingInfo{},
+		agents:         make(map[*actor.Ref]*agentState),
+		taskList:       newTaskList(),
+		groups:         make(map[*actor.Ref]*group),
+		queuePositions: make(jobSortState),
```

Comment:
I'm not sure I understand the point of having a queuePositions map here. What does this bring that groups doesn't already have? This also seems like it requires us to create the groupActorToID map just to translate from actor.Ref and access queuePositions.

Reply:
hmm it might be easier to do a call but some of my thinking is:
1. we want RM to persist and restore this scheduling state but nothing else from the group and having them as a map together makes it simpler
2. the other is that direct missing jobid to group mapping I mentioned in the other comment